src/VisualStudio/Core/Def/Implementation/ExtractInterface/VisualStudioExtractInterfaceOptionsService.cs

```diff
@@ -51,16 +52,29 @@ public VisualStudioExtractInterfaceOptionsService(IGlyphService glyphService)
 
             if (result.HasValue && result.Value)
             {
+                var includedMembers = viewModel.MemberContainers.Where(c => c.IsChecked).Select(c => c.MemberSymbol);
+
                 return new ExtractInterfaceOptionsResult(
                     isCancelled: false,
-                    includedMembers: viewModel.MemberContainers.Where(c => c.IsChecked).Select(c => c.MemberSymbol),
+                    includedMembers: includedMembers.AsImmutable(),
                     interfaceName: viewModel.InterfaceName.Trim(),
-                    fileName: viewModel.FileName.Trim());
+                    fileName: viewModel.FileName.Trim(),
+                    location: GetLocation(viewModel.Destination));
             }
             else
             {
                 return ExtractInterfaceOptionsResult.Cancelled;
             }
         }
+
+        private static ExtractInterfaceOptionsResult.ExtractLocation GetLocation(InterfaceDestination destination)
+        {
+            switch (destination)
+            {
+                case InterfaceDestination.CurrentFile: return ExtractInterfaceOptionsResult.ExtractLocation.SameFile;
```

Comment:
is there a reason why it has 2 different enum or const? can same thing used all places

Reply:
The enum definition has to be accessible for XAML bindings. While we have `InternalsVisibleTo` in the binary itself, there are parts generated by XAML that doesn't apply to. This was the only way to do it without either hacking so the XAML stuff has `InternalsVisibleTo` somehow, or making a public API change.